libraries/cmake/source/modules/utils.cmake

```diff
@@ -31,10 +31,19 @@ function(initializeGitSubmodule submodule_path no_recursive shallow)
     set(optional_depth_arg "")
   endif()
 
+  # In git versions >= 2.18.0 we need to explicitly set the protocol
+  # in order to do a shallow clone without error.
+  if(GIT_VERSION_STRING VERSION_EQUAL "2.18.0" OR GIT_VERSION_STRING VERSION_GREATER "2.18.0")
```

Comment:
I like this. To work around the gitlab issue. do we need another condition? Maybe on shallow?

Reply:
I don't think we can work around the GitLab issue. I think this depends on their git server's configuration. It might improve in the future but the protocol-2-approach we're using has some gaps due to these server expectations.

Reply:
My understanding is the issue is only present for a shallow clone. So this workaround is only needed for the shallow cloning.

Reply:
BTW, on my home not-very-fast cable modem
```
$ time git clone --quiet https://gitlab.com/cryptsetup/cryptsetup.git

real	0m4.690s
user	0m0.925s
sys	0m0.536s
```

From real network, it's sub 1 second.

Reply:
Testing, I guess you can leave the `-c protocol.version=2` in place even for gitlab. It just wouldn't help, but it doesn't cause an error. 

If that matches your understanding, I think I'm good with this. If there are performance issues due to the cloning, we should tackle them separately, possible by mirroring to github.